### PR TITLE
refactor: improve if ternary ENV testing

### PR DESCRIPTION
**Pull request by rmdhfz**
**Description**
Refactor code structure and consolidate similar sections for improved readability

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: app/Config/Database.php [hunk 1]
```diff
@@ -74,12 +74,9 @@ class Database extends Config
     public function __construct()
     {
         parent::__construct();
-
```
Comment (kenjis) [inline app/Config/Database.php:77]: Thank you for trying to improve CI4!

But I like the empty line.

File: app/Config/Database.php [hunk 2]
```diff
@@ -74,12 +74,9 @@ class Database extends Config
...
         // Ensure that we always set the database group to 'tests' if
         // we are currently running an automated test suite, so that
         // we don't overwrite live data on accident.
-        if (ENVIRONMENT === 'testing') {
-            $this->defaultGroup = 'tests';
-        }
+        $this->defaultGroup = (ENVIRONMENT === 'testing') ? 'tests' : 'default';
```
Comment (kenjis) [inline app/Config/Database.php:80]: This breaks existing apps, because the `defaultGroup` may not be `default`.

**Comment by kenjis** [comment]
Please sign all your commits.

> We ask that contributions have code commits signed. This is important in order to prove, as best we can, the provenance of contributions.
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/signing.md

**Comment by kenjis** [comment]
It is not recommended to include merge commits in PR branches. 
If you can use git, use the `git rebase` command. 
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#updating-your-branch

**Comment by paulbalandan** [comment]
I believe the existing code is fine.

**Comment by kenjis** [comment]
Unfortunately, we don't think this is an improve. 

If you want to try to contribute to CI4, try good first issues: https://github.com/codeigniter4/CodeIgniter4/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

Thank you!

**Pull request closed without merging**